### PR TITLE
add the conway sequence

### PR DESCRIPTION
**Pull request by eloibru**
- [ ] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Java/blob/master/CONTRIBUTING.md).
- [ ] This pull request is all my own work -- I have not plagiarized it.
- [ ] All filenames are in PascalCase.
- [ ] All functions and variable names follow Java naming conventions.
- [ ] All new algorithms have a URL in their comments that points to Wikipedia or other similar explanations.

### CODE REVIEW

**Review by siriak — CHANGES_REQUESTED** [review]
The code looks good, could you please add some JUnit tests? (see https://github.com/TheAlgorithms/Java/tree/master/src/test/java/com/thealgorithms)

**Comment by eloibru** [comment]
Done, I've added a few tests.

File: src/main/java/com/thealgorithms/others/Conway.java [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    * This class will generate the conway sequence also known as the look and say sequence n (I just added a prompt to ask the user the starting point of the sequence).
+    * This works with any characters (of course it's better with numbers)
+    * To generate a member of the sequence from the previous member, read off the digits of the previous member, counting the number of digits in groups of the same digit. For example:
+    *1 is read off as "one 1" or 11.
+    *11 is read off as "two 1s" or 21.
+    *21 is read off as "one 2, one 1" or 1211.
+    *1211 is read off as "one 1, one 2, two 1s" or 111221.
+    *111221 is read off as "three 1s, two 2s, one 1" or 312211.
+    * https://en.wikipedia.org/wiki/Look-and-say_sequence
+    * */
```
Comment (siriak) [inline src/main/java/com/thealgorithms/others/Conway.java:17]:
```suggestion

    /*
    * This class will generate the conway sequence also known as the look and say sequence.
    * To generate a member of the sequence from the previous member, read off the digits of the previous member, counting the number of digits in groups of the same digit. For example:
    *1 is read off as "one 1" or 11.
    *11 is read off as "two 1s" or 21.
    *21 is read off as "one 2, one 1" or 1211.
    *1211 is read off as "one 1, one 2, two 1s" or 111221.
    *111221 is read off as "three 1s, two 2s, one 1" or 312211.
    * https://en.wikipedia.org/wiki/Look-and-say_sequence
    * */
```

File: src/main/java/com/thealgorithms/others/Conway.java [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+    public static void main(String[]args) {
+        Scanner scanner;
+        String originalString;
+        System.out.println("Enter any character you want : ");
+        scanner = new Scanner(System.in);originalString = scanner.nextLine();
+
+        List<String> strings = generateList(originalString);
+        System.out.println(strings);
+
+    }
```
Comment (siriak) [inline src/main/java/com/thealgorithms/others/Conway.java:29]:
```suggestion
```

File: src/main/java/com/thealgorithms/others/Conway.java [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+    public static String generateNextElement(String originalString) {
+
+        builder.setLength(0);
+        String[] stp = originalString.split("(?<=(.))(?!\\1)");
+        Arrays.stream(stp).forEach(s -> builder.append(s.length()).append(s.charAt(0)));
+        return builder.toString();
+    }
+
+
+
```
Comment (siriak) [inline src/main/java/com/thealgorithms/others/Conway.java:50]:
```suggestion
```

File: src/main/java/com/thealgorithms/others/Conway.java [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+        for(int i=0; i<20; i++) {
+            originalString =   generateNextElement(originalString);
+            numbers.add(originalString);
+            }
+        return numbers;
+    }
+
+
+    public static String generateNextElement(String originalString) {
+
```
Comment (siriak) [inline src/main/java/com/thealgorithms/others/Conway.java:42]:
```suggestion
```

File: src/main/java/com/thealgorithms/others/Conway.java [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+        System.out.println(strings);
+
+    }
+
+    protected static List<String> generateList(String originalString) {
+        List<String> numbers = new ArrayList<>();
+        for(int i=0; i<20; i++) {
+            originalString =   generateNextElement(originalString);
+            numbers.add(originalString);
+            }
```
Comment (siriak) [inline src/main/java/com/thealgorithms/others/Conway.java:36]:
```suggestion
            originalString = generateNextElement(originalString);
            numbers.add(originalString);
        }
```

File: src/test/java/com/thealgorithms/others/ConwayTest.java [hunk 6]
```diff
@@ -0,0 +1,40 @@
+package com.thealgorithms.others;
+
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+public class ConwayTest {
+
+
```
Comment (siriak) [inline src/test/java/com/thealgorithms/others/ConwayTest.java:10]:
```suggestion
```

File: src/test/java/com/thealgorithms/others/ConwayTest.java [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+    @Test
+    public void testGenerateWith1(){
+        assertTrue(Conway.generateList("1").contains("1321132132211331121321231231121113112221121321132122311211131122211211131221131211132221121321132132212321121113121112133221123113112221131112311332111213122112311311123112111331121113122112132113213211121332212311322113212221"));
```
Comment (siriak) [inline src/test/java/com/thealgorithms/others/ConwayTest.java:13]: Please check a specific element of the sequence. For example, the last one

File: src/main/java/com/thealgorithms/others/Conway.java [hunk 8]
```diff
@@ -0,0 +1,51 @@
...
+        scanner = new Scanner(System.in);originalString = scanner.nextLine();
+
+        List<String> strings = generateList(originalString);
+        System.out.println(strings);
+
+    }
+
+    protected static List<String> generateList(String originalString) {
+        List<String> numbers = new ArrayList<>();
+        for(int i=0; i<20; i++) {
```
Comment (siriak) [inline src/main/java/com/thealgorithms/others/Conway.java:33]: What is `20`? Please make this a function parameter

**Review by siriak — CHANGES_REQUESTED** [review]

File: src/main/java/com/thealgorithms/others/Conway.java [hunk 9]
```diff
@@ -0,0 +1,57 @@
...
+
+        do {
+        System.out.println("Enter the maximum of iteration (max allowed 50) : ");
+        scanner = new Scanner(System.in);
+        maxIteration = scanner.nextInt();
+        }
+        while (maxIteration < MIN_ITERATION_ALLOWED || maxIteration > MAX_ITERATION_ALLOWED);
+        List<String> strings = generateList(originalString, maxIteration);
+        System.out.println(strings);
+    }
```
Comment (siriak) [inline src/main/java/com/thealgorithms/others/Conway.java:39]:
```suggestion
```

**Review by siriak — APPROVED** [review]
Looks good, thanks!

**Pull request merged**